xline/src/storage/kvstore.rs

```diff
@@ -237,7 +237,46 @@ impl KvStoreInner {
             count: kvs.len().cast(),
             ..RangeResponse::default()
         };
-        response.kvs = kvs;
+        if !req.count_only {
```

Comment:
Please check how etcd handle this.

Reply:
[etcd](https://github.com/etcd-io/etcd/blob/main/server/etcdserver/txn/txn.go#L166-L205)  
here is the code for etcd to handle this. I think it's more complicated than the current implementation